packages/terra-application-layout/src/header/_ApplicationHeader.jsx

```diff
@@ -0,0 +1,253 @@
+   */
+  applicationLinks: ApplicationLinksPropType,
+  /**
+   * The element to be placed within the fit start area for extensions within the layout.
+   */
+  extensions: PropTypes.element,
+  /**
+   * The Object of layout-related APIs provided to the components of the Layout.
+   */
+  layoutConfig: ApplicationLayoutPropTypes.layoutConfigPropType,
+  /**
+   * The set of routes currently identified by the NavigationLayout. This prop is provided by the NavigationLayout.
+   */
+  navigationLayoutRoutes: PropTypes.arrayOf(processedRoutesPropType),
+  /**
+   * The window size currently identified by the NavigationLayout. This prop is provided by the NavigationLayout.
+   */
+  navigationLayoutSize: PropTypes.string,
+  /**
+   * Configuration values for the ApplicationName component.
+   */
+  nameConfig: ApplicationLayoutPropTypes.nameConfigPropType,
+  /**
+   * Configuration to be provided to the ApplicationUtility component.
+   */
+  utilityConfig: ApplicationLayoutPropTypes.utilityConfigPropType,
+  /**
+   * Internationalization object with translation APIs. Provided by `injectIntl`.
+   */
+  intl: intlShape,
+};
+
+class ApplicationHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleUtilityDiscloseRequest = this.handleUtilityDiscloseRequest.bind(this);
+    this.handleUtilityPopupCloseRequest = this.handleUtilityPopupCloseRequest.bind(this);
+    this.handleUtilityOnChange = this.handleUtilityOnChange.bind(this);
+    this.getTargetRef = this.getTargetRef.bind(this);
+    this.setContentNode = this.setContentNode.bind(this);
+    this.renderToggle = this.renderToggle.bind(this);
+    this.renderAppName = this.renderAppName.bind(this);
+    this.renderExtensions = this.renderExtensions.bind(this);
+    this.renderNavigation = this.renderNavigation.bind(this);
+    this.renderUtilities = this.renderUtilities.bind(this);
+    this.renderUtilitiesPopup = this.renderUtilitiesPopup.bind(this);
+
+    this.state = { utilityComponent: undefined };
+  }
+
+  setContentNode(node) {
+    this.contentNode = node;
+  }
+
+  getTargetRef() {
+    if (this.contentNode) {
+      return this.contentNode.querySelector('[data-application-header-utility]');
+    }
+    return undefined;
+  }
+
+  handleUtilityDiscloseRequest(utility) {
+    this.setState({
+      utilityComponent: React.cloneElement(utility, { onRequestClose: this.handleUtilityPopupCloseRequest }),
+    });
+  }
+
+  handleUtilityPopupCloseRequest() {
+    if (this.state.utilityComponent) {
+      this.setState({ utilityComponent: undefined });
+    }
+  }
+
+  handleUtilityOnChange(event, key) {
+    const { utilityConfig, app } = this.props;
+
+    utilityConfig.onChange(event, key, app && app.disclose);
+  }
+
+  renderToggle() {
+    const { layoutConfig, intl } = this.props;
+
+    if (layoutConfig.toggleMenu) {
+      return (
+        <div className={cx('toolbar-toggle')}>
+          <button
```

Comment:
Is it possible to use the `terra-button`? We have lost consistent keyboard navigation here

Reply:
Might have to defer to @dkasper-was-taken here, but I think the most recent terra-button variant updates have left us with no options to render the toggle button appropriately.

If there are navigation differences, we'll have to fix them here.

Reply:
Hmm I would be curiuos to see if we could align with terra-button because there is visual inconsistencies for focus/active styling of buttons with firefox.